assets/js/theme/common/product-details.js

```diff
@@ -591,4 +592,41 @@ export default class ProductDetails extends ProductDetailsBase {
             detail: { productDetails },
         }));
     }
+
+    updateData() {
+        $(document).ready(() => {
+            const monthSelector = $('#month-selector');
+            const daySelector = $('#day-selector');
+            const yearSelector = $('#year-selector');
+
+            const updateDays = () => {
+                const month = parseInt(monthSelector.val(), 10);
+                const year = parseInt(yearSelector.val(), 10);
+                let daysInMonth;
+
+                if (!Number.isNaN(month) && !Number.isNaN(year)) {
+                    switch (month) {
+                    case 2:
+                        daysInMonth = (year % 4 === 0 && year % 100 !== 0) || (year % 400 === 0) ? 29 : 28;
```

Comment:
@VladDlogush as I understand this line we basically need to check a leap year and if so return 29days for February. What's the reason behind  checking for `year % 100` or `year % 400` ?

Reply:
@bc-alexsaiannyi this are rules how leap year can be defined. Leap year - this year it is included in 4, but not included in 100 unless it is included in 400

Reply:
@yurytut1993 is right! This is the rule for determining a leap year
@bc-alexsaiannyi